src/rust/src/declarative_asn1/types.rs

```diff
@@ -108,6 +116,116 @@ impl PrintableString {
     }
 }
 
+pub(crate) struct DateTimeInfo {
+    pub(crate) datetime: asn1::DateTime,
+    pub(crate) nanoseconds: Option<u32>,
+}
+
+// Internal utility function used to extract data from a Python datetime
+// object. Used by both GeneralizedTime and UtcTime.
+pub(crate) fn get_datetime_info(
```

Comment:
This should be organized next to `py_to_datetime` and have an analogous name. I don't think we need a `DateTimeInfo` type -- a tuple is fine.

Reply:
fixed